### PR TITLE
Allow to specify KubeObjectDetailRegistration priority

### PR DESCRIPTION
**Pull request by jakolehm**
Allows extensions (or core) to specify placement of object details. For example:

```ts
  kubeObjectDetailItems = [
    {
      kind: "Pod",
      apiVersions: ["v1"],
      priority: 10,
      components: {
        Details: (props: any) => <CustomPodDetails pod={props.object} />
      }
    }
  ]
```
Would be shown above Pod events.

### CODE REVIEW

File: src/renderer/components/+network-ingresses/ingress-details.tsx [hunk 1]
```diff
@@ -141,3 +139,11 @@ kubeObjectDetailRegistry.add({
     Details: (props) => <IngressDetails {...props} />
   }
 })
+kubeObjectDetailRegistry.add({
+  kind: "Ingress",
+  apiVersions: ["extensions/v1beta1"],
```
Comment (nevalla) [inline src/renderer/components/+network-ingresses/ingress-details.tsx:144]: should we register also `networking.k8s.io/v1`?

Comment (jakolehm) [reply src/renderer/components/+network-ingresses/ingress-details.tsx:144]: Isn't that out-of-scope of this PR?

Comment (nevalla) [reply src/renderer/components/+network-ingresses/ingress-details.tsx:144]: probably since we are using `extensions/v1beta1` in api too.

Comment (jakolehm) [reply src/renderer/components/+network-ingresses/ingress-details.tsx:144]: `networking.k8s.io/v1` should be done in a separate PR, it most likely will affect also how UI should be rendered.

**Review by nevalla — APPROVED** [review]
LGTM

File: src/renderer/components/+network-ingresses/ingress-details.tsx (hunk 1 above)
Comment (jakolehm) [reply src/renderer/components/+network-ingresses/ingress-details.tsx:144]: Created an issue: https://github.com/lensapp/lens/issues/1347

**Pull request merged**